server/socket/socket.js

```diff
@@ -0,0 +1,25 @@
+const { io } = require("../app");
+const jwt = require("jsonwebtoken");
+const cookie = require('cookie');
+
+const getDecodedToken = (socket) => {
+  const cookies = cookie.parse(socket.handshake.headers.cookie);
+  const token = cookies.token;
+  const decodedToken = jwt.verify(token, process.env.SECRET);
+  return decodedToken;
+}
+
+io.use((socket, next) => {
+  if (getDecodedToken(socket)) {
```

Comment:
oh, what I meant is that maybe you should associate the user id with the socket in this middleware

Reply:
Alright, I changed it to put the userId in the middleware